app/models/batch_connect/session.rb

```diff
@@ -77,8 +77,14 @@ def db_root
       # @return [Array<Session>] list of sessions
       def all
         db_root.children.select(&:file?).map do |f|
-          new.from_json(f.read)
-        end.map do |s|
+          begin
+            new.from_json(f.read)
+          rescue => e
+            Rails.logger.error("ERROR: Error parsing file '#{f}' --- #{e.class} - #{e.message}")
+            f.delete unless Rails.env.development?
```

Comment:
Auto clean up may be nice but could be annoying in production for sites debugging problems as a corrupted session file would disappear. When can a session file become corrupted to produce this error?

Reply:
If it isn't cleaned up then logs will become full of error messages for every corrupt file.

> When can a session file become corrupted to produce this error?

Right now only if the Linux kernel is interrupted during the write process (e.g., due to quota limit being reached or power outage). Or if the user goes in there and edits the files by hand.